src/ActiveBackground.tsx

```diff
@@ -1,18 +1,34 @@
-import * as React from 'react'
+import type { ReactNode } from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
 import { useResizeObserver } from './hooks/useResizeObserver'
-import { IActiveBackground, IActiveBackgroundPattern } from './index'
 
-export const ActiveBackground: React.FC<IActiveBackground> = ({
-  pattern,
-  patternOptions = {},
-  className,
-  children,
-}) => {
-  const [canvasRef, setRef] = React.useState<HTMLCanvasElement | null>(null)
-  const onRefSet = React.useCallback((ref) => setRef(ref), [setRef])
-  const [update, setUpdate] = React.useState({})
+export type PatternOptions = Record<string, unknown>
+export interface PatternConstructor<TOptions> {
+  new (
+    canvas: HTMLCanvasElement,
+    options?: TOptions
+  ): ActiveBackgroundPattern
+}
+export interface ActiveBackgroundPattern {
+  render(): void
+  start(): void
+  stop(): void
+}
+
+export interface ActiveBackgroundProps<TPatternOptions = PatternOptions> {
+  Pattern: PatternConstructor<TPatternOptions>
+  patternOptions?: TPatternOptions
+  className?: string
+  children: ReactNode
+}
+
+
+export function ActiveBackground({ Pattern, patternOptions, className, children }: ActiveBackgroundProps): JSX.Element {
```

Comment:
Just my preferred style

Reply:
that's reasonable, i just let prettier do its thing

Reply:
Oh yeah, I didn't mean that putting it on one line was my preferred style, but using a function instead of a `const ActiveBackground` and only explicitly saying this function never returns (renders) `null`, instead of `React.XX<YY>.